llama-index-core/llama_index/core/workflow/context.py

```diff
@@ -40,6 +83,83 @@ def __init__(self, workflow: "Workflow", stepwise: bool = False) -> None:
         # Step-specific instance
         self._events_buffer: Dict[Type[Event], List[Event]] = defaultdict(list)
 
+        # keep track of all the event classes that are accepted by the workflow
+        self._event_classes: Dict[str, Type[Event]] = {}
+        for step_func in self._workflow._get_steps().values():
+            step_config: Optional[StepConfig] = getattr(
+                step_func, "__step_config", None
+            )
+            if step_config and step_config.accepted_events:
+                for event_type_cls in step_config.accepted_events:
+                    self._event_classes[event_type_cls.__name__] = event_type_cls
+
+            if step_config and step_config.return_types:
+                for return_type_cls in step_config.return_types:
+                    self._event_classes[return_type_cls.__name__] = return_type_cls
+
+    def _serialize_event(self, event: Event) -> Tuple[str, Any]:
+        return (event.__class__.__name__, event.model_dump())
+
+    def _deserialize_event(self, event_name: str, event_data: Any) -> Event:
+        event_cls = self._event_classes[event_name]
+        return event_cls.model_validate(event_data)
```

Comment:
Can we move these into the `Event` class?

Reply:
ah good call